netbox/resource_netbox_config_context.go

```diff
@@ -0,0 +1,514 @@
+
+	platforms := res.GetPayload().Platforms
+	platformsSlice := make([]int64, len(platforms))
+	for i, v := range platforms {
+		platformsSlice[i] = int64(v.ID)
+	}
+	d.Set("platforms", platformsSlice)
+
+	regions := res.GetPayload().Regions
+	regionsSlice := make([]int64, len(regions))
+	for i, v := range regions {
+		regionsSlice[i] = int64(v.ID)
+	}
+	d.Set("regions", regionsSlice)
+
+	roles := res.GetPayload().Roles
+	rolesSlice := make([]int64, len(roles))
+	for i, v := range roles {
+		rolesSlice[i] = int64(v.ID)
+	}
+	d.Set("roles", rolesSlice)
+
+	site_groups := res.GetPayload().SiteGroups
+	siteGroupsSlice := make([]int64, len(site_groups))
+	for i, v := range site_groups {
+		siteGroupsSlice[i] = int64(v.ID)
+	}
+	d.Set("site_groups", siteGroupsSlice)
+
+	sites := res.GetPayload().Sites
+	sitesSlice := make([]int64, len(sites))
+	for i, v := range sites {
+		sitesSlice[i] = int64(v.ID)
+	}
+	d.Set("sites", sitesSlice)
+
+	tags := res.GetPayload().Tags
+	tagsSlice := make([]string, len(tags))
+	for i, v := range tags {
+		tagsSlice[i] = string(v)
+	}
+	d.Set("tags", tagsSlice)
+
+	tenant_groups := res.GetPayload().TenantGroups
+	tenantGroupsSlice := make([]int64, len(tenant_groups))
+	for i, v := range tenant_groups {
+		tenantGroupsSlice[i] = int64(v.ID)
+	}
+	d.Set("tenant_groups", tenantGroupsSlice)
+
+	tenants := res.GetPayload().Tenants
+	tenantsSlice := make([]int64, len(tenants))
+	for i, v := range tenants {
+		tenantsSlice[i] = int64(v.ID)
+	}
+	d.Set("tenants", tenantsSlice)
+
+	return nil
+}
+
+func resourceNetboxConfigContextUpdate(d *schema.ResourceData, m interface{}) error {
+	api := m.(*client.NetBoxAPI)
+
+	id, _ := strconv.ParseInt(d.Id(), 10, 64)
+
+	/*name := d.Get("name").(string)
+	datajson := d.Get("data").(string)
+	cluster_groups := d.Get("cluster_groups").([]int64)
+	cluster_types := d.Get("cluster_types").([]int64)
+	clusters := d.Get("clusters").([]int64)
+	device_types := d.Get("device_types").([]int64)
+	locations := d.Get("locations").([]int64)
+	platforms := d.Get("platforms").([]int64)
+	regions := d.Get("regions").([]int64)
+	roles := d.Get("roles").([]int64)
+	site_groups := d.Get("site_groups").([]int64)
+	sites := d.Get("sites").([]int64)
+	tenant_groups := d.Get("tenant_groups").([]int64)
+	tenants := d.Get("tenants").([]int64)
+	tags := d.Get("tags").([]string)
+
+	weightValue, weightOk := d.GetOk("weight")
+	var weight int64
+	// Default weight to 1000 if not given
+	if !weightOk {
+		weight = 1000
+	} else {
+		weight = weightValue.(int64)
+	}
+
+	data.Weight = &weight
+	data.Name = &name
+	data.Data = &datajson
+	data.ClusterGroups = cluster_groups
+	data.ClusterTypes = cluster_types
+	data.Clusters = clusters
+	data.DeviceTypes = device_types
+	data.Locations = locations
+	data.Platforms = platforms
+	data.Regions = regions
+	data.Roles = roles
+	data.SiteGroups = site_groups
+	data.Sites = sites
+	data.TenantGroups = tenant_groups
+	data.Tenants = tenants
+	data.Tags = tags
+	*/
```

Comment:
?

Reply:
Oops, forgot to delete my old code! Nothing to see here!